### PR TITLE
removes declared hooks on componentWillUnmount

### PR DESCRIPTION
**Pull request by lukaszsobek**
Associated Issue: #3822

### Summary of Changes

* Removal of listeners on componentWillUnmount

### Test Plan

Tested the memory footprint by repeatedly opening and closing a tab 

### Screenshots/Videos (OPTIONAL)
Shows memory consumption before the change, and with 2 variations of changes implemented (the fix contains variation 2). 

http://prntscr.com/gyziln

### CODE REVIEW

File: src/components/Editor/Preview/index.js [hunk 1]
```diff
@@ -39,16 +39,29 @@ class Preview extends PureComponent {
     const self = this;
     self.onScroll = this.onScroll.bind(this);
     self.onMouseOver = debounce(this.onMouseOver, 40);
+    self._onMouseOver = e => this.onMouseOver(e);
+    self._onMouseUp = e => this.onMouseUp(e);
+    self._onMouseDown = e => this.onMouseDown(e);
```
Comment (jasonLaster) [inline src/components/Editor/Preview/index.js]: can we bind here: `self.onMouseDown = this.onMouseDown.bind(this)`

**Comment by jasonLaster** [comment]
What does this show us:



what is the current result? is there a leak, if so ... how much?

File: src/components/Editor/Preview/index.js (hunk 1 above)
Comment (lukaszsobek) [reply src/components/Editor/Preview/index.js]: Sure. Will update.

**Comment by jasonLaster** [comment]
@lukaszsobek thanks for the detailed write up. 

Let's dig a bit:

1. Source documents has a map of code mirror docs. Do we add to the map?
2. Redux has a store of sources. Does the redux sources store change?

**Comment by lukaszsobek** [comment]
@jasonLaster Here is another screenshot set comparing the memory - this time with snapshorts of editor window closed as well. All images are about the same debugger session on the same window with just opening and closing the main.js editor panel (notice how the memory use increases on strings and when the window gets closed http://prntscr.com/h0rx6w

**Comment by wldcordeiro** [comment]
Looks like this is ready to roll? From my read-through @lukaszsobek has responded to review and this looks like it gives us an improvement.

**Pull request merged**